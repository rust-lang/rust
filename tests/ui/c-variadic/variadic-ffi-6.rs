#![crate_type="lib"]
#![feature(c_variadic)]

pub unsafe extern "C" fn use_vararg_lifetime(
    x: usize,
    y: ...
) -> &usize { //~ ERROR missing lifetime specifier
    &0
}

pub unsafe extern "C" fn use_normal_arg_lifetime(x: &usize, y: ...) -> &usize { // OK
    x
}
