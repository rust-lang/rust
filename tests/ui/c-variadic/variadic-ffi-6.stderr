error[E0106]: missing lifetime specifier
  --> $DIR/variadic-ffi-6.rs:7:6
   |
LL | ) -> &usize {
   |      ^ expected named lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but there is no value for it to be borrowed from
help: consider using the `'static` lifetime, but this is uncommon unless you're returning a borrowed value from a `const` or a `static`
   |
LL | ) -> &'static usize {
   |       +++++++
help: instead, you are more likely to want to change one of the arguments to be borrowed...
   |
LL |     x: &usize,
   |        +
help: ...or alternatively, you might want to return an owned value
   |
LL - ) -> &usize {
LL + ) -> usize {
   |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0106`.
